### PR TITLE
Handle basex, basey, linthreshx, linthreshy deprecated in matplotlib 3.3

### LINKED ISSUE #3181: [BUG] More PyPlot matplotlib 3.3 deprecations
## Details
```
using Plots
pyplot()
plot(rand(10), xscale = :log10, yscale = :log10)
```
gives
```
sys:1: MatplotlibDeprecationWarning: The 'basex' parameter of __init__() has been renamed 'base' since Matplotlib 3.3; support for the old name will be dropped two minor releases later.
sys:1: MatplotlibDeprecationWarning: The 'linthreshx' parameter of __init__() has been renamed 'linthresh' since Matplotlib 3.3; support for the old name will be dropped two minor releases later.
sys:1: MatplotlibDeprecationWarning: The 'basey' parameter of __init__() has been renamed 'base' since Matplotlib 3.3; support for the old name will be dropped two minor releases later.
sys:1: MatplotlibDeprecationWarning: The 'linthreshy' parameter of __init__() has been renamed 'linthresh' since Matplotlib 3.3; support for the old name will be dropped two minor releases later.
```

Earlier, some Matplotlib 3.3 deprecations were handled in https://github.com/JuliaPlots/Plots.jl/issues/2874 .

### Backends

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |     |     |
pyplot       | x    |     |
plotly       |     |     |
plotlyjs     |     |     |
pgfplotsx    |     |     |
inspectdr    |     |     |

### Versions

Plots.jl version: v1.9.1
Backend  version (`]st -m`): PyPlot v2.9.0
Output of `versioninfo()`:
```
Julia Version 1.6.0-DEV.1511
Commit 85000b154b (2020-11-17 22:54 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.0 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = "C:\Program Files\Sublime Text 3\sublime_text.exe"
  JULIA_NUM_THREADS = 4
```

### PR DESCRIPTION
**Pull request by jaakkor2**
See https://matplotlib.org/3.3.0/api/api_changes.html

Fixes #3181 .

### CODE REVIEW

File: src/backends/pyplot.jl [hunk 1]
```diff
@@ -799,6 +799,9 @@ function py_set_scale(ax, sp::Subplot, axis::Axis)
     letter = axis[:letter]
     scale in supported_scales() || return @warn("Unhandled scale value in pyplot: $scale")
     func = getproperty(ax, Symbol("set_", letter, "scale"))
+    if PyPlot.version ≥ v"3.3" # https://matplotlib.org/3.3.0/api/api_changes.html
```
Comment (isentropic) [inline src/backends/pyplot.jl:802]: Would you please send me the permalink to the title of this exact pyplot change?

**Review by isentropic — CHANGES_REQUESTED** [review]

File: src/backends/pyplot.jl (hunk 1 above)
Comment (isentropic) [reply src/backends/pyplot.jl:802]: I could not find the relevant news myself

Comment (isentropic) [reply src/backends/pyplot.jl:802]: Nevermind I found it now.

**Review by isentropic — APPROVED** [review]

**Comment by isentropic** [comment]
Would you please run a few quick tests with axes scales and post the output here? Just a few your own tests, that would include setting `(x,y,z)scale=:log10, log2` and seeing if stuff still works. So, I expect some code and output pngs and we are ready to merge this I think

**Comment by jaakkor2** [comment]
Copying the relevant passage from https://matplotlib.org/3.3.0/api/api_changes.html
>log/symlog scale base, ticks, and nonpos specification¶

>semilogx, semilogy, loglog, LogScale, and SymmetricalLogScale used to take keyword arguments that depends on the axis orientation ("basex" vs "basey", "subsx" vs "subsy", "nonposx" vs "nonposy"); these parameter names are now deprecated in favor of "base", "subs", "nonpositive". This deprecation also affects e.g. ax.set_yscale("log", basey=...) which must now be spelled ax.set_yscale("log", base=...).

>The change from "nonpos" to "nonpositive" also affects LogTransform, InvertedLogTransform, SymmetricalLogTransform, etc.

>To use different bases for the x-axis and y-axis of a loglog plot, use e.g. ax.set_xscale("log", base=10); ax.set_yscale("log", base=2).

**Comment by jaakkor2** [comment]
```
using Plots
pyplot()
plot(rand(10), xscale = :log10, yscale = :log2)
```
Figure_1
```
plot(rand(10), xscale = :ln, yscale = :log10)
```
Figure_1

**Comment by isentropic** [comment]
Thanks for the PR

**Comment by isentropic** [comment]
Perhaps you'd be interested in https://github.com/JuliaPlots/Plots.jl/issues/3171 and https://github.com/JuliaPlots/Plots.jl/issues/3168

**Pull request merged**